File: backend/schemas/auth/response.py
from pydantic import BaseModel
from uuid import UUID
from models.role import Role


class CreatedUserResponse(BaseModel):
    username: str
    email: str


class UserResponse(BaseModel):
    user_id: UUID
    user_name: str
    email: str
    role: str
    is_active: bool
    created_at: str
    updated_at: str | None

class TokenResponse(BaseModel):
    access_token: str
    token_type: str
    expired_at: str
    current_user: UserResponse

File: frontend/src/app/schema/auth/response.ts
export type UserResponse = {
    id: number;
    user_name: string;
    email: string;
    is_active: boolean;
    role: string
    created_at: string;
    updated_at: string;
};

export type CreatedUserResponse = {
    username: string;
    email: string;
}
